model/src/main/java/org/projectnessie/model/Validation.java

```diff
@@ -25,11 +25,14 @@
 
 /** Collection of validation rules. */
 public final class Validation {
-  public static final String HASH_REGEX = "^[0-9a-fA-F]{8,64}$";
-  public static final String REF_NAME_REGEX =
-      "^[A-Za-z](((?![.][.])[A-Za-z0-9./_-])*[A-Za-z0-9._-])?$";
+
+  public static final String HASH_RAW_REGEX = "[0-9a-fA-F]{8,64}";
+  public static final String HASH_REGEX = "^" + HASH_RAW_REGEX + "$";
+  public static final String REF_NAME_RAW_REGEX =
+      "[A-Za-z](((?![.][.])[A-Za-z0-9./_-])*[A-Za-z0-9._-])";
+  public static final String REF_NAME_REGEX = "^" + REF_NAME_RAW_REGEX + "?$";
   public static final String REF_NAME_OR_HASH_REGEX =
-      "^(([0-9a-fA-F]{8,64})|([A-Za-z](((?![.][.])[A-Za-z0-9./_-])*[A-Za-z0-9._-])?))$";
+      "^((" + HASH_RAW_REGEX + ")|(" + REF_NAME_RAW_REGEX + "?))$";
```

Comment:
Could the `?` be moved into `REF_NAME_RAW_REGEX` for the sake of clarity? It looks like it always follows it anyway.

Reply:
Nice catch! Yea - that's where it actually should be.